src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -150,7 +157,15 @@ private void initialize() {
     this.add(listShowPanel, listPanelC);
     this.add(buttonPanel, buttonPanelC);
     this.add(currentPanel, currentPanelC);
-    //TextValidator validator = new TextValidator();
+
+    //load existing users from database
+    con = new Connector();
+    List<String> usersToAdd = con.getAllUsers();
+    for(String s : usersToAdd) {
+      clientContext.user.addUser(s);
```

Comment:
nit (this was original code): Having clientContext.user a public field is gross (EJ item 14).